Samples/Catalog/Sources/AlertController.swift

```diff
@@ -19,12 +19,94 @@
 import Foundation
 import UIKit
 
+extension UIAlertAction {
+  
+  /// Action types most commonly used
+  public enum actionType{
+    
+    ///Ok Option
+    case ok
+    
+    /// Default Cancel Option
+    case cancel
+    
+    /// Destructive action with custom title
+    case destructive(String)
+    
+    /// Custom action with title and style
+    case custom(String,UIAlertActionStyle)
+    
+    /**
+     Creates the action instance for UIAlertController
+     - parameter handler: Call Back function
+     - returns UIAlertAction Instance
+     */
+    public func action(handler:((String) -> Void)? = nil) -> UIAlertAction {
+      
+      //Default value
+      var actionStyle = UIAlertActionStyle.default
+      var title = ""
+      
+      // Action configuration based on the action type
+      switch self {
+        
+      case .cancel:
+        actionStyle = .cancel
+        title = "Cancel"
+        
+      case .destructive(let optionTitle):
+        title = optionTitle
+        actionStyle = .destructive
+        
+      case .custom(let optionTitle, let style):
+        title = optionTitle
+        actionStyle = style
+        
+      default:
+        title = "Ok"
+      }
+      
+      //Creating UIAlertAction instance
+      let action = UIAlertAction(title:title,style:actionStyle, handler: {(nativeAction) in
+        if (handler != nil){
```

Comment:
This can be simplified and be more Swifty using the following:
```
if let handler = handler {
  handler(title)
}
```
Or even `handler.map({ $0() })` (the latter is less preferred version)